### PR TITLE
[Topology] Fix method isQuadDeulaunayOriented and its use in splitAlongPath

### PR DESCRIPTION
**Pull request by epernod**
Method isQuadDeulaunayOriented  was only handling some specific case. Use directly the computation of angles to check flipping Delaunay condition to triangulate a quad see [wikipedia page](https://en.wikipedia.org/wiki/Delaunay_triangulation#Visual_Delaunay_definition:_Flipping) 

And in fact the method was not working well but the method splitAlongPath used for triangular incision was calling it with wrong quad structure. 
Luckily in this case we had error*error = good behavior... 

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/TriangleSetGeometryAlgorithms.inl [hunk 1]
```diff
@@ -944,44 +944,28 @@ bool TriangleSetGeometryAlgorithms< DataTypes >::isQuadDeulaunayOriented(const t
...
+    sofa::type::Vec<3, Real> CD = { p4[0] - p3[0], p4[1] - p3[1], p4[2] - p3[2] };
+
+    AB.normalize();
+    AD.normalize();
+    Real alpha = acos(dot(AB, AD));
+
+    CB.normalize();
+    CD.normalize();
+    Real beta = acos(CB * CD);
+    bool isDelau = (alpha + beta > M_PI) ? false : true;
```
Comment (alxbilger) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/TriangleSetGeometryAlgorithms.inl]:
```suggestion
    const bool isDelau = (alpha + beta <= M_PI);
```

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/TriangleSetGeometryAlgorithms.inl [hunk 2]
```diff
@@ -944,44 +944,28 @@ bool TriangleSetGeometryAlgorithms< DataTypes >::isQuadDeulaunayOriented(const t
...
+    //std::cout << "alpha: " << alpha*180/ M_PI << " | beta: " << beta * 180 / M_PI << " delaunay: " << isDelau << std::endl;
```
Comment (alxbilger) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/TriangleSetGeometryAlgorithms.inl]:
```suggestion
```

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/TriangleSetGeometryAlgorithms.inl [hunk 3]
```diff
@@ -3651,16 +3637,16 @@ int TriangleSetGeometryAlgorithms<DataTypes>::SplitAlongPath(PointID pa, Coord&
                 // Triangularize the remaining quad according to the delaunay criteria
                 if (isQuadDeulaunayOriented(posOrder[0], posOrder[1], posOrder[2], posOrder[3]))
                 {
-                    new_triangles.push_back(Triangle(vertexOrder[1], vertexOrder[2], vertexOrder[3]));
+                    new_triangles.push_back(Triangle(vertexOrder[1], vertexOrder[2], vertexOrder[4]));
```
Comment (alxbilger) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/TriangleSetGeometryAlgorithms.inl]:
```suggestion
                    new_triangles.emplace_back(vertexOrder[1], vertexOrder[2], vertexOrder[4]);
```

**Review by alxbilger — CHANGES_REQUESTED** [review]

**Comment by bakpaul** [comment]
One unit test failling : 
https://ci.inria.fr/sofa-ci-dev/job/sofa-framework/job/PR-4494/CI_CONFIG=ubuntu_gcc,CI_PLUGINS=options,CI_TYPE=release/2/testReport/UnitTests.Sofa_Component/Topology_InciseProcessor_test/InciseTriangles/

**Comment by epernod** [comment]
> One unit test failling : https://ci.inria.fr/sofa-ci-dev/job/sofa-framework/job/PR-4494/CI_CONFIG=ubuntu_gcc,CI_PLUGINS=options,CI_TYPE=release/2/testReport/UnitTests.Sofa_Component/Topology_InciseProcessor_test/InciseTriangles/

Unit test references have been changed. The new version is more accurate.

**Pull request merged**